### ISSUE TITLE
Can't OpenWith ROMs with RetroArch

### ISSUE THREAD

**Issue by amirrammari** [libretro/RetroArch#9718]
Hello, I can't open with ROMs with RetroArch.
Please add this feature.
`retroarch.exe "D:\Castle of Illusion Starring Mickey Mouse (USA, Europe).smd"`

**Comment by amirrammari** [comment]
Also not working below command:
`C:\Users\Amir\AppData\Roaming\RetroArch\retroarch.exe C:\Users\Amir\AppData\Roaming\RetroArch\cores\genesis_plus_gx_libretro.dll "D:\Amir\Games\ROMs\Sega - Mega Drive - Genesis\Castle of Illusion Starring Mickey Mouse (USA, Europe).smd"`

**Comment by RobLoach** [comment]
The problem I see here is determining what the intended core to load is. There was also this discussion to have `retroarch.exe` work without passing the `-L` core option: https://github.com/libretro/RetroArch/issues/746

**Comment by amirrammari** [comment]
Cores should be loaded based on ROMs file extension.
For example, all .smd files (Sega Mega Drive) loaded with Genesis Plus GX.
```
C:\Users\Amir\AppData\Roaming\RetroArch\retroarch.exe -L C:\Users\Amir\AppData\Roaming\RetroArch\cores\genesis_plus_gx_libretro.dll "%1"
```
![3E0F8EA2-2664-4781-B56D-B150BDFE6F17](https://user-images.githubusercontent.com/23485114/68844599-4eef4a00-06df-11ea-8232-154c3376d5c5.GIF)

**Comment by n1ghttrap** [comment]
Seconding this.  The only solution for now is for the end user to painstakingly make a series of custom association strings in the 3rd party Default Programs Editor.  What's worse is that if you associate a lot of filetypes with "retroarch.exe" and then edit the association in DPE, I notice there's some sort of overlap and associations are lost as you add more.  One way to get around this is to associate the filetypes with dummy executables from Explorer, and then edit them to face retroarch.exe in Default Programs Editor with the right core via -L and "%1".  It's a HUGE pain.

Retroarch loading the only core capable of running the ROM file type, or in the case of multiple cores asking you which one you'd like to launch, on association would be EXTREMELY convenient.  Thanks in advance.

**Comment by fr500** [comment]
It's not possible. The extension is not enough because SEVERAL CORES may support the same extension.

**Comment by n1ghttrap** [comment]
> It's not possible. The extension is not enough because SEVERAL CORES may support the same extension.

Could it not just ask which one you want to use in that case?  This is exactly what it does when you pick an extension from the file browser.  The thing is, the file browser is not nearly as good as Windows is at quickly reaching a particular file.  It would be preferable to cut the middle man, if possible.

**Comment by n1ghttrap** [comment]
Alternatively, a search-within-subdirectories would suffice.  Either within the Desktop Menu or the Retroarch File Browser.  That way one could search for Mario and get results for several systems, for instance.  Better still, would be a random selection option, but I know that's a bit of a fluffy extra.  Regardless, these things can be done easily within Windows Explorer, and an easier way to associate files to Retroarch would be nice.  Thanks for the consideration.

**Comment by n1ghttrap** [comment]
For those wanting this functionality in the likely event this gets closed:

Make a dummy exe file for EACH file extension you want to run.  Otherwise things will get overwritten and won't stick.  Like nes.exe for .nes, etc. Associate the filetype with it in explorer so the filetype is in the registry.  Then use Default Programs Editor https://defaultprogramseditor.com and open the filetype, then edit the Open commend with the following program:  "C:\retroarch\retroarch.exe" -L "C:\retroarch\cores\core.dll" "%1" replacing the directory/drive with what applies to your machine.  It's annoying, but possible to do. 

Hopefully a simple association to Retroarch will be possible some day.

**Comment by n1ghttrap** [comment]
Another alternative these days: you can just drag files into an open RA session and it'll ask which core to use or load the only one capable of playing it automatically.  Simply associating files to RA and having them open the right core automatically, or with a prompt, would still be nicer.

**Comment by Maingron** [comment]
For me, I've now settled with creating a bunch of bat-files in a folder which then access a .lnk shortcut to Retroarch.  
Not beautiful, but at least it's fairly maintainable, I guess.

Example `Retroarch - NES.bat`
```bat
start retroarch.lnk -L "mesen" %1
exit
```

Example `Retroarch - SNES.bat`
```bat
start retroarch.lnk -L "snes9x" %1
exit
```

And then just set Open With accordingly

**Comment by Maingron** [comment]
So, in addition to my previous comment, I've now created a script for that: https://github.com/Maingron/Open-with-Retroarch/tree/main

**Comment by BenMcLean** [comment]
I do not believe any of the people who say that this isn't possible. This is absolutely possible and quite easy to do in fact. We just need one of the devs to shut up and do it or accept a PR that does.

If multiple cores support a file extension, show a dialog to choose a core, just like the file browser within RetroArch does. It is not a big ask to let Windows users have this and does not involve any secret proprietary Microsoft functionality.

**Comment by RobLoach** [comment]
This Pull Request will really help.  Would love some eyes on it: https://github.com/libretro/RetroArch/pull/15458

**Comment by BenMcLean** [comment]
Related issue #9275

**Comment by BenMcLean** [comment]
Might be nice to add a feature to automatically register all those file associations as something that could be selected in the RetroArch menu.
